### PR TITLE
Use variable for App name in translations

### PR DESCRIPTION
**Pull request by Qstick**
#### Database Migration
NO

#### Description
Replace "Sonarr" with a variable for translations, this allows translation key sharing in Weblate with other apps without clobbering translations with the wrong app name.

### CODE REVIEW

File: src/NzbDrone.Core/Localization/Core/cs.json [hunk 1]
```diff
@@ -5,21 +5,21 @@
     "AgeWhenGrabbed": "Stáří (kdy bylo získáno)",
     "Always": "Vždy",
     "AnalyseVideoFiles": "Analyzovat video soubory",
-    "AnalyseVideoFilesHelpText": "Extrahujte ze souborů informace o videu, jako je rozlišení, doba běhu a informace o kodeku. To vyžaduje, aby Sonarr četl části souboru, což může způsobit vysokou aktivitu disku nebo sítě během skenování.",
+    "AnalyseVideoFilesHelpText": "Extrahujte ze souborů informace o videu, jako je rozlišení, doba běhu a informace o kodeku. To vyžaduje, aby {appName} četl části souboru, což může způsobit vysokou aktivitu disku nebo sítě během skenování.",
     "ApplicationURL": "URL aplikace",
     "ApplicationUrlHelpText": "Externí adresa URL této aplikace včetně http(s)://, portu a základní adresy URL",
-    "AuthenticationMethodHelpText": "Vyžadovat uživatelské jméno a heslo pro přístup k Sonarru",
+    "AuthenticationMethodHelpText": "Vyžadovat uživatelské jméno a heslo pro přístup k {appName}u",
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Localization/Core/cs.json:11]: /bSonarr/

Comment (Qstick) [reply src/NzbDrone.Core/Localization/Core/cs.json:11]: Nah, need to do all languages to fix things or they may never get changed.

File: frontend/src/Utilities/String/translate.ts [hunk 2]
```diff
@@ -27,7 +27,9 @@ export default function translate(
   key: string,
   tokens?: Record<string, string | number | boolean>
 ) {
-  const translation = translations[key] || key;
+  let translation = translations[key] || key;
+
+  translation = translation.replace(/\{(appName)\}/gi, 'Sonarr');
```
Comment (markus101) [inline frontend/src/Utilities/String/translate.ts:32]: I don't think we should replace tokens outside of the normal means, I'm not sure we need to pass in `appName` in every situation this is used, but should use tokens to replace it.

**Review by markus101 — CHANGES_REQUESTED** [review]

File: frontend/src/Utilities/String/translate.ts (hunk 2 above)
Comment (Qstick) [reply frontend/src/Utilities/String/translate.ts:32]: Your're going to make me work for this one I see :D

Comment (markus101) [reply frontend/src/Utilities/String/translate.ts:32]: Hey, I gave you a shortcut to just add it to tokens if it's not there 😄

**Review by markus101 — APPROVED** [review]

**Pull request merged**